drivers/usb/gadget/xaptum/f_psock_gadget.c

```diff
@@ -224,23 +327,87 @@ static struct usb_gadget_strings *psock_strings[] = {
  **********************************************************************/
 
 
+
+void psock_debug_hex_dump (char *desc, void *addr, int len)
+{
+    int i;
+    unsigned char buff[17];
+    unsigned char *pc = (unsigned char*)addr;
+
+    // Output description if given.
+    if (desc != NULL)
+        printk ("%s (%d byte read):\n", desc, len);
+
+    if (len == 0)
+    {
+        printk("  ZERO LENGTH\n");
+        return;
+    }
+    if (len < 0)
+    {
+        printk("  NEGATIVE LENGTH: %i\n",len);
+        return;
+    }
+
+    // Process every byte in the data.
+    for (i = 0; i < len; i++)
+    {
+        // Multiple of 16 means new line (with line offset).
+
+        if ((i % 16) == 0)
+        {
+            // Just don't print ASCII for the zeroth line.
+            if (i != 0)
+                printk (KERN_CONT "  %s\n", buff);
+
+            // Output the offset.
+            printk (KERN_CONT "  %04x ", i);
+        }
+
+        // Now the hex code for the specific character.
+        printk (KERN_CONT " %02x", pc[i]);
+
+        // And store a printable ASCII character for later.
+        if ((pc[i] < 0x20) || (pc[i] > 0x7e))
+            buff[i % 16] = '.';
+        else
+            buff[i % 16] = pc[i];
+        buff[(i % 16) + 1] = '\0';
+    }
+
+    // Pad out last line if not exactly 16 characters.
+    while ((i % 16) != 0)
+    {
+        printk (KERN_CONT "   ");
+        i++;
+    }
+
+    // And print the final ASCII bit.
+    printk (KERN_CONT "  %s\n", buff);
+}
+
+
 /**
  * usb allocation
  */
 
 static inline struct f_psock *func_to_psock(struct usb_function *f)
 {
+	printk(KERN_INFO "func_to_psock enter");
         return container_of(f, struct f_psock, function);
 }
 
 
+/* Binds this driver to a device */
```

Comment:
This functions sets up the endpoints and successfully runs before the crash. The following is outputted

```
psock_bind enter
Configured cmd_out_ep
Configured cmd_in_ep
psock_bind exit ret=0
```